### PR TITLE
Merge QT4 Migration into Master

### PR DESCRIPTION
**Pull request by tst-cjeffries**
Merges all of the QT4 work into master.

To test (on all supported platforms, including EL8):

1. Build and install OpenGEE.
2. Download and build all tutorial data.
3. Verify that in looks correct in EC and on the server admin page.
4. Create custom databases from the tutorial databases.
5. Verify that it looks correct in EC and on the server admin page.
6. Test historical data, search, and the various other plugins.

### CODE REVIEW

**Comment by tst-ccamp** [comment]
I started testing on a fresh CentOS 7 system. I followed the instructions in the build setup file and it fails the png check: `Checking for C library gtest... yes
Checking for C header file libpng12/png.h... no
libpng12 must be installed!
scons: *** [build] Error 1
scons: building terminated because of errors.
`

Looks like that check needs to be updated or removed.

**Comment by tst-dsugar** [comment]
@googlebot I consent.

**Comment by tst-mswartz** [comment]
@googlebot I consent.

File: earth_enterprise/src/fusion/mtjtools/scroll/application.cpp [hunk 1]
```diff
@@ -12,36 +12,48 @@
...
+#include <Qt/q3simplerichtext.h>
+#include <Qt/q3simplerichtext.h>
+#include <Qt/q3grid.h>
+#include <Qt/q3canvas.h>
+//#include <qsimplerichtext.h>
+//#include <qgrid.h>
+//#include <qobject.h>
+//#include <qcanvas.h>
+#include <Qt/qobject.h>
+#include  <Qt/q3filedialog.h>
```
Comment (tst-basilhuffman) [inline earth_enterprise/src/fusion/mtjtools/scroll/application.cpp:44]: remove commented out code

File: earth_enterprise/src/fusion/mtjtools/scroll/application.h [hunk 2]
```diff
@@ -17,6 +17,9 @@
 #ifndef APPLICATION_H
 #define APPLICATION_H
 
+//#include <qobject.h>
+#include <QObject>
+//#include <qobject.h>
```
Comment (tst-basilhuffman) [inline earth_enterprise/src/fusion/mtjtools/scroll/application.h:22]: remove commented out code

File: earth_enterprise/src/fusion/mtjtools/scroll/imageview.h [hunk 3]
```diff
@@ -19,7 +19,7 @@
 #ifndef __IMAGE_VIEW_H__
 #define __IMAGE_VIEW_H__
 
-#include <qscrollview.h>
+#include <Qt3Support/Q3ScrollView>//q3scrollview.h>
```
Comment (tst-basilhuffman) [inline earth_enterprise/src/fusion/mtjtools/scroll/imageview.h:22]: remove commented out code

**Review by tst-basilhuffman — COMMENTED** [review]
mostly just removing commented out code, a couple other questions that need clarification

**Review by tst-ccamp — APPROVED** [review]

**Comment by tst-ccamp** [comment]
Tested this on CentOS 7 and Ubuntu 16. You have my approval.

File: earth_enterprise/src/common/khxml/khidl.pl [hunk 4]
```diff
@@ -1753,7 +1760,7 @@ sub EmitDOMReader
 	print $fh "{\n";
     } else {
 	print $fh "void\n";
-	print $fh "FromElement(DOMElement *elem, $class->{qualname} &self)\n";
+  print $fh "FromElement(DOMElement *elem, $class->{qualname} &self)\n";
```
[the remark this answers is not in the document]
Comment (tst-dsugar) [reply earth_enterprise/src/common/khxml/khidl.pl:1762]: I found so many other indent issues...it seemed best to skip this for now.

File: earth_enterprise/src/fusion/autoingest/sysman/SConscript [hunk 5]
```diff
@@ -107,65 +107,61 @@ sysmanFiles = ['AssetD.cpp', 'AssetVersionD.cpp',
...
+     env['BUILDDATE'],
+     ],
+    LIBS=['geautoingest', 'geconfigutil', 'gefconfigutil', 'geutil', 'gecommon', 'gexml', 'gemiscconfig', 'xerces-c', 'QtCore', 'QtGui'])
 
 
 env.install('fusion_bin', [ gesystemmanager, geresourceprovider ] )
 env.install('fusion_root', [ 'gefusion' ], 'etc/init.d')
 
 
+'''
```
[the remark this answers is not in the document]
Comment (tst-dsugar) [reply earth_enterprise/src/fusion/autoingest/sysman/SConscript:139]: This is being marked with a TODO to evaluate post merge...

File: earth_enterprise/src/fusion/fusionui/SConscript [hunk 6]
```diff
@@ -339,11 +345,11 @@ fusion = env.executable('fusion', objs + qtobjs + idlobjs)
...
-def buildtrans(env, target, source):
-    cmd = '%s %s -qm %s' % ( env['LRELEASE'], str(source[0].name),
-                             str(target.name) )
-    print cmd, '\n'
-    return os.system(cmd)
+#def buildtrans(env, target, source):
+#    cmd = '%s %s -qm %s' % ( env['LRELEASE'], str(source[0].name),
+#                             str(target.name) )
+#    print cmd, '\n'
+#    return os.system(cmd)
```
[the remark this answers is not in the document]
Comment (tst-dsugar) [reply earth_enterprise/src/fusion/fusionui/SConscript:352]: This is documentary for how to do translations...

File: earth_enterprise/src/fusion/fusionui/SystemManager.cpp [hunk 7]
```diff
@@ -47,10 +50,16 @@ class WaitingItem : public QListBoxText
   }
 };
 
-class WaitingErrorTip : public QToolTip
+/* Momentarily disabled
```
[the remark this answers is not in the document]
Comment (tst-dsugar) [reply earth_enterprise/src/fusion/fusionui/SystemManager.cpp]: Marked as TODO

File: earth_enterprise/src/fusion/fusionui/main.cpp [hunk 8]
```diff
@@ -194,14 +194,14 @@ int main(int argc, char** argv) {
...
-  QTranslator translator(0);
+  // disabling for now
+  /*QTranslator translator(0,0);
   if (translator.load(QString("fusion_") + QTextCodec::locale(), ".")) {
     a.installTranslator(&translator);
   } else if (translator.load(QString("fusion_") + QTextCodec::locale(),
                              kGESharePath)) {
     a.installTranslator(&translator);
-  }
+  }*/
```
[the remark this answers is not in the document]
Comment (tst-dsugar) [reply earth_enterprise/src/fusion/fusionui/main.cpp:199]: This is a TODO

File: earth_enterprise/src/fusion/fusionui/spacer.h [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+#define SPACER_H
+
+#include <Qt/qlayoutitem.h>
+
+class spacer : public QSpacerItem
+{
+public:
+    spacer() = default;
+
+    void setObjectName(const QString& oName)
```
[the remark this answers is not in the document]
Comment (tst-dsugar) [reply earth_enterprise/src/fusion/fusionui/spacer.h]: I think so...

**Review by tst-basilhuffman — APPROVED** [review]

**Pull request merged**